File: code/c++/insertion_sort.cpp
#include<iostream>
#include<vector>

std::vector<int> insertion_sort(std::vector<int>);
int main()
{
    std::vector<int> list {7,1,2,3,4,3,6,7,4,3};
    list=insertion_sort(list);
    for(int i=0;i<list.size();i++)
    {
        std::cout<<list[i]<<" ";
    }
    std::cout<<std::endl;
    return 0;
}
std::vector<int> insertion_sort(std::vector<int> arr)
{
    for(int i=1;i<arr.size();i++)
    {
        for(int j=i-1;j>=0;j--)
        {
            if (arr[j]>arr[j+1])
            {
                int temp=arr[j];
                arr[j]=arr[j+1];
                arr[j+1]=temp;
            }
            else
            {
                break;
            }
        }
    }
    return arr;
}

File: code/python/insertion_sort.py
def insertion_sort(arr):
    length=len(arr)
    for i in range(1,length):
        for j in range(i):
            if (arr[i-1-j]>arr[i-j]):
                arr[i-1-j],arr[i-j]=arr[i-j],arr[i-1-j]                
    return arr

print(insertion_sort([1,0]))
